core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -1797,12 +1811,29 @@ public void doUpdateRoleListOfUserWithID(String userID, String[] deletedRoles, S
                     if (sqlStmt1 == null) {
                         throw new UserStoreException("The sql statement for remove user from role is null.");
                     }
+                    if (StringUtils.isBlank(sqlStmt3) && isUniqueGroupIdEnabled()) {
+                        throw new UserStoreException("The sql statement for update group last modified time is null.");
+                    }
                     if (sqlStmt1.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
                         DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt1, roles, tenantId, userID,
                                 tenantId, tenantId);
                     } else {
                         DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt1, roles, userID);
                     }
+
+                    // If there are common groups in new and deleted, their modified time should only be updated
```

Comment:
Shall we explain the reason more clearly?.  
When the same group is add and removed from a user, the total effect is not a change for that corresponding group, hence skipping last modified time update.

Reply:
Addressed with 9deb4f3c8436cd4f60f7f84cccb53f18dc0cd1a2